### DISCUSSION TITLE
Equivalent to OpenCV's remap

### DISCUSSION THREAD

**Discussion by nullpainter** [SixLabors/ImageSharp#1312] [Q&A] [answered]
Hi everyone,

Sorry if this is a RTFM question, but I couldn't see anything - does ImageSharp have an arbitrary (x,y) -> (x', y') transformation method similar to OpenCV's remap?

In other words, apply a general transformation from the _target_ image back to the source, interpolating values as appropriate? I could do the interpolation myself, but it seems like something which should already exist.

Cheers!

M

**Comment by nullpainter** [comment]
Or, if it doesn't have a drop-in equivalent to remap, is there just a fractional interpolation mechanism? i.e., so requesting a pixel at non-integer coordinates results in an appropriately interpolated value?

**Comment by antonfirsov** [comment, marked as answer]
@nullpainter we only support Affine and Projective matrix transforms out of the box, but it has interpolation and resampling.

If I understand it correctly remap would mean an arbitrary transformation defined by a function or some sort of pixel map array right?

**Reply by nullpainter** [reply]
Yes, that's correct. My specific use case is transforming raw satellite images of the Earth into arbitrary map projections.

I've built it by hand, but even just a basic fractional pixel interpolation method would be really handy (e.g., "get me the interpolated RGBA value in my image at pixel position [1.2, 3.5]").

**Reply by antonfirsov** [reply]
Unfortunately, I can only point you to obscure internal code. Method [`LinearTransformUtilities.Convolve`](https://github.com/SixLabors/ImageSharp/blob/8e2792e3ab96dd69d8e39255e83afdee493b1b26/src/ImageSharp/Processing/Processors/Transforms/Linear/LinearTransformUtilities.cs#L31) is being used to do convolution using a resampler and some temporary buffers provided in a parameters. [Here is](https://github.com/SixLabors/ImageSharp/blob/master/src/ImageSharp/Processing/Processors/Transforms/Linear/AffineTransformProcessor%7BTPixel%7D.cs#L210) how the code is being used.

Not sure if it helps though :\

**Reply by nullpainter** [reply]
That's a great pointer - thank you!
